msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -685,7 +685,7 @@ internal MsalAccountCacheItem GetAccountCacheItem(MsalAccountCacheKey msalAccoun
             InstanceDiscoveryMetadataEntry instanceDiscoveryMetadata = null;
 
             var authorityType = Authority.GetAuthorityType(authority);
-            if (authorityType == Core.Instance.AuthorityType.Aad || authorityType == Core.Instance.AuthorityType.B2C)
+            if (authorityType == Core.Instance.AuthorityType.Aad)
             {
                 instanceDiscoveryMetadata = await AadInstanceDiscovery.Instance.GetMetadataEntryAsync(
```

Comment:
>AadInstanceDiscovery.Instance [](start = 50, length = 29)

this is actually the type imo, which should be set to a B2C Instance discovery. It seems odd to have to if here... the correct type should be injected